source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -207,45 +217,60 @@ void *ns_dyn_mem_alloc(int16_t alloc_size)
                     //Free
                     if (s_size >= alloc_sector) {
                         /*found block*/
+
                         if (s_size > (alloc_sector + 4)) {
                             //debug("Bigger\r\n");
                             //Set sector start len & Move pointer to Allocated data sector
                             *ptr = alloc_sector;
-                            //Move Pointer  to next free place
-                            ptr -= alloc_sector;
-                            //Set return Pointer
-                            retval = (void *) ptr;
-                            ptr--;
-                            //Set End length
-                            *ptr-- = alloc_sector;
+                            if (dir_up) {
```

Comment:
Seems a bit laboured, especially compared to the "hole"-creating bit below. 

Can't you actually unify the two cases anyway - at the minute it's 

    if (making hole) {
          make block
          make hole
    } else {
          make block
    }

Would make sense to avoid repeating the block creation. Can't quite write it out, but something like

   if (space for hole) {
         create hole (now at front, rather than back, because it's easier)
    }
   else {
        increase alloc_sector to take extra space
   }
   create block